### PR TITLE
support multi-line calls in sprintf() again

### LINKED ISSUE #1249: `sprintf_linter()` errors for multi-line `sprintf()` statements with comments
`Error in parsed_expr[[2L]] (sprintf_linter.R#24): subscript out of bounds`

This is because `xml2::xml_text()` doesn't respect newlines and thus
```r
sprintf(
  "asd", # comment
  b
)
```

when run through `xml_text()` turns into `sprintf("asd", # commentb)`, which is not parseable.

### PR DESCRIPTION
**Pull request by AshesITR**
fixes #1249
added a regression test

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -238,3 +238,36 @@ get_r_string <- function(s, xpath = NULL) {
...
+#'
+#' @param xml An `xml_node`.
+#'
+#' @return A source-code equivalent of `xml` with unnecessary whitespace removed.
+#'
+#' @noRd
+get_r_code <- function(xml) {
+  # shortcut if xml has line1 and line2 attrs and they are equal
+  # if they are missing, xml_attr() returns NA, so we continue
+  if (isTRUE(xml2::xml_attr(xml, "line1") == xml2::xml_attr(xml, "line2"))) {
```
Comment (MichaelChirico) [inline R/utils.R:254]: this makes it non-vectorized, where `xml_text()` was earlier -- is that intentional?

File: R/utils.R [hunk 2]
```diff
@@ -238,3 +238,36 @@ get_r_string <- function(s, xpath = NULL) {
...
+    return(xml2::xml_text(xml))
+  }
+  # find all unique line numbers of atoms (i.e. expressions
```
Comment (MichaelChirico) [inline R/utils.R:257]: just missing `)` or anything else?

File: R/utils.R (hunk 1 above)
Comment (AshesITR) [reply R/utils.R:254]: `capture_sprintf_warning()` is already non-vectorized, so it's fine.

File: R/utils.R (hunk 2 above)
Comment (AshesITR) [reply R/utils.R]: actually a relic of an older approach which wouldn't have worked with multi-line strings.

**Comment by MichaelChirico** [comment]
would it be simpler to back up to using the `source_expression$lines` in this case once we know the line/column where the `sprintf()` call starts/stops

**Comment by AshesITR** [comment]
We have a `get_content()`. That would work for all nodes with full location info (line1, line2, col1, col2).
I'll see if there is an easy way to use that.

**Comment by MichaelChirico** [comment]
Worth adding a TODO instead?

**Pull request merged**